src/main/java/net/mcreator/blockly/java/blocks/TextIndexOfBlock.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2024, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.blockly.java.blocks;
+
+import net.mcreator.blockly.BlocklyCompileNote;
+import net.mcreator.blockly.BlocklyToCode;
+import net.mcreator.blockly.IBlockGenerator;
+import net.mcreator.generator.template.TemplateGeneratorException;
+import net.mcreator.ui.init.L10N;
+import net.mcreator.util.XMLUtil;
+import org.w3c.dom.Element;
+
+import java.util.List;
+
+public class TextIndexOfBlock implements IBlockGenerator {
+
+	@Override public void generateBlock(BlocklyToCode master, Element block) throws TemplateGeneratorException {
+		List<Element> elements = XMLUtil.getDirectChildren(block);
+		Element check = null, text = null, from = null;
+		for (Element element : elements)
+			if (element.getNodeName().equals("value"))
+				if (element.getAttribute("name").equals("check"))
+					check = element;
+				else if (element.getAttribute("name").equals("from"))
+					from = element;
+				else if (element.getAttribute("name").equals("text"))
+					text = element;
+
+		if (text != null && check != null && from != null) {
+			master.append("(");
+			master.processOutputBlock(text);
+			master.append(".indexOf(");
+			master.processOutputBlockWithoutParentheses(check);
+			master.append(",");
+			master.processOutputBlockWithoutParentheses(from);

```

Comment:
The reason for the change suggested earlier was so as not to generate extra code when not needed, maybe like this:
```suggestion
			if (!XMLUtil.getChildrenWithName(from, "block").isEmpty()) {
				master.append(",");
				master.processOutputBlockWithoutParentheses(from);
			}
```

Reply:
This will not work because shadow block can still contain non-0 value that should be considered.

Reply:
For this reason I suggested going with two blocks too.

Reply:
I don't see it being worth adding another block for the sake of avoiding a number 0 in the generated code. If this was the main reason, it may be better to not use shadow block then at all